app/services/user_alerts/alert_user_about_new_device.rb

```diff
@@ -59,10 +59,7 @@ def self.sign_in_events_start_time(user:)
       # Typically, it's guaranteed that even in the worst-case of a sign-in occurring immediately
       # after a scheduled job run, it should take no longer than twice the scheduled delay. A small
       # buffer is added to account for delays of the job run or within the job itself.
-      [
-        user.sign_in_new_device_at,
-        (IdentityConfig.store.new_device_alert_delay_in_minutes * 3).minutes.ago,
-      ].max
+      (IdentityConfig.store.new_device_alert_delay_in_minutes * 1.5).minutes.ago
```

Comment:
Can you explain what the fix is?

Reply:
It takes the delay in minutes (5 minutes) and multiplies by 1.5 and sets that as the beginning of the date range of events that would get included in the email assembled on line 30. I experimented with different timing and that fosters the expected outcome in the bug ticket. This seems to give just enough buffer.
It sends any interim failed attempts but not the initial sign in like you flagged here: https://github.com/18F/identity-idp/pull/10659#discussion_r1609931766

Reply:
I feel like this is not actually getting to the root of the issue, which is that the initial email will [unset `sign_in_new_device_at`](https://github.com/18F/identity-idp/blob/5442bbdb9af5421dcb4fabc7a8ec4ce9ab307526/app/services/user_alerts/alert_user_about_new_device.rb#L51), and while we'll [send an alert](https://github.com/18F/identity-idp/blob/5442bbdb9af5421dcb4fabc7a8ec4ce9ab307526/app/controllers/concerns/two_factor_authenticatable_methods.rb#L23) once they fully authenticate, the failed attempts won't be reflected since [`sign_in_new_device` is only set at that point in time](https://github.com/18F/identity-idp/blob/5442bbdb9af5421dcb4fabc7a8ec4ce9ab307526/app/controllers/concerns/two_factor_authenticatable_methods.rb#L19-L21).

I think what we should do instead is either assign `sign_in_new_device_at` any time someone MFAs (successful or not), or change the logic of [assigning `sign_in_new_device_at` in the valid authentication](https://github.com/18F/identity-idp/blob/5442bbdb9af5421dcb4fabc7a8ec4ce9ab307526/app/controllers/concerns/two_factor_authenticatable_methods.rb#L19-L21) to find the time of the earliest failed authentication since the last email was sent.

Reply:
Also, can we add a regression test case here? I'm not sure if the current test changes are expected or help validate the changes.